octobot_trading/exchanges/parser/funding_rate_parser.py

```diff
@@ -0,0 +1,151 @@
+from octobot_trading.enums import ExchangeConstantsFundingColumns as FundingCols
+from octobot_trading.exchanges.parser.util import Parser
+import octobot_trading.exchanges.parser.util as parser_util
+from octobot_trading import constants as trading_constants
+from octobot_commons import constants as commons_constants
+
+
+class FundingRateParser(Parser):
+    """
+    overwrite FundingRateParser class methods if necessary
+    always/only include bulletproof custom code into the parser to improve generic support
+
+        parser usage:   parser = FundingRateParser(exchange)
+                        funding_rate = parser.parse_funding_rate(raw_funding_rate)
+                        funding_rates = parser.parse_funding_rate_list(raw_funding_rates)
+    """
+    
+    FUNDING_TIME_UPDATE_PERIOD = 8 * commons_constants.HOURS_TO_SECONDS
+
+    def __init__(self, exchange):
+        super().__init__(exchange=exchange)
+        self.PARSER_TITLE = "funding rate"
+
+    def parse_funding_rate_list(self, raw_funding_rates: list) -> list:
+        """
+        use this method to format a list of funding rates
+        :param raw_funding_rates: raw funding rate list
+        :return: formatted funding rate list
+        """
+        self._ensure_list(raw_funding_rates)
+        self.formatted_records = (
+            [
+                self.parse_funding_rate(raw_funding_rate)
+                for raw_funding_rate in raw_funding_rates
+            ]
+            if self.raw_records
+            else []
+        )
+        self._create_debugging_report()
+        return self.formatted_records
+
+    def parse_funding_rate(self, raw_funding_rate: dict, from_ticker=False):
+        """
+        use this method to parse a raw funding rate
+        :param raw_funding_rate:
+        :return: formatted funding rate
+        """
+        self._ensure_dict(raw_funding_rate)
+        try:
+            if from_ticker:
+                self.PARSER_TITLE = "funding rate from ticker"
+            else:
+                self._parse_timestamp()
+                self._parse_symbol()
+            self._parse_funding_rate()
+            self._parse_predicted_funding_rate()
+            self._parse_next_funding_time()
+            self._parse_last_funding_time()
+
+        except Exception as e:
+            # just in case something bad happens
+            self._log_missing(
+                "funding rate parser broken",
```

Comment:
I think we can just say to the user "failed to parse funding rate. Please contact support"

Reply:
i changed it